src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            Debug.Assert(node != null);
+
+            if (node.LeftOperand is null)
+            {
+                if (node.RightOperand is null)
+                {
+                    // Case 1:              ".."
+                    // translates to:       Range.All()
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__All);
+                    return MakeCall(node.Syntax, rewrittenReceiver: null, method, ImmutableArray<BoundExpression>.Empty, node.Type);
+                }
+
+                BoundExpression rightOperand = VisitExpression(node.RightOperand);
+                rightOperand = NullableAlwaysHasValue(rightOperand) ?? rightOperand;
+
+                if (node.RightOperand.Type.IsNullableType())
+                {
+                    // Case 2:              "..nullableRight"
+                    // translates to:       nullableRight.HasValue ? new Nullable(Range.ToEnd(nullableRight.GetValueOrDefault())) : default
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd);
+                    return LiftRangeExpression(node.Syntax, node.Type, method, rightOperand);
+                }
+                else
+                {
+                    // Case 3:              "..right"
+                    // translates to:       Range.ToEnd(right)
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd);
+                    ImmutableArray<BoundExpression> arguments = ImmutableArray.Create(rightOperand);
+                    return MakeCall(node.Syntax, rewrittenReceiver: null, method, arguments, node.Type);
+                }
+            }
+            else
+            {
+                BoundExpression leftOperand = VisitExpression(node.LeftOperand);
+                leftOperand = NullableAlwaysHasValue(leftOperand) ?? leftOperand;
+
+                if (node.RightOperand is null)
+                {
+                    if (leftOperand.Type.IsNullableType())
+                    {
+                        // Case 4:              "nullableLeft.."
+                        // translates to:       nullableLeft.HasValue ? new Nullable(Range.FromStart(nullableLeft.GetValueOrDefault())) : default
+
+                        MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__FromStart);
+                        return LiftRangeExpression(node.Syntax, node.Type, method, leftOperand);
+                    }
+                    else
+                    {
+                        // Case 5:              "left.."
+                        // translates to:       Range.FromStart(left)
+
+                        MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__FromStart);
```

Comment:
>MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__FromStart); [](start = 24, length = 113)

It feels like this can be moved out of the ```if``` and ```else```.

Reply:
I prefer to keep this for readability reasons.

---
In reply to: [185365846](https://github.com/dotnet/roslyn/pull/26405#discussion_r185365846) [](ancestors = 185365846)